app/assets/javascripts/discourse/app/services/modal.js

```diff
@@ -3,28 +3,69 @@ import { getOwner } from "@ember/application";
 import I18n from "I18n";
 import { dasherize } from "@ember/string";
 import { disableImplicitInjections } from "discourse/lib/implicit-injections";
+import { tracked } from "@glimmer/tracking";
 
 @disableImplicitInjections
 export default class ModalService extends Service {
   @service appEvents;
 
+  @tracked name;
+  @tracked opts = {};
+  @tracked selectedPanel;
+  @tracked hidden = true;
+
+  @tracked titleOverride;
+  @tracked modalClassOverride;
+  @tracked onSelectPanel;
+
+  get title() {
+    if (this.titleOverride) {
+      return this.titleOverride;
+    } else if (this.opts.titleTranslated) {
+      return this.opts.titleTranslated;
+    } else if (this.opts.title) {
+      return I18n.t(this.opts.title);
+    } else {
+      return null;
+    }
+  }
+
+  set title(value) {
+    this.titleOverride = value;
+  }
+
+  get modalClass() {
+    if (!this.#isRendered) {
+      return null;
+    }
+
+    if (this.modalClassOverride) {
+      return this.modalClassOverride;
+    } else {
+      return (
+        this.opts.modalClass ||
+        `${dasherize(this.name.replace(/^modals\//, "")).toLowerCase()}-modal`
+      );
+    }
+  }
+
+  set modalClass(value) {
+    this.modalClassOverride = value;
+  }
+
   show(name, opts = {}) {
     const container = getOwner(this);
     const route = container.lookup("route:application");
-    const modalController = route.controllerFor("modal");
 
-    modalController.set(
-      "modalClass",
-      opts.modalClass || `${dasherize(name).toLowerCase()}-modal`
-    );
+    this.opts = opts;
 
     const controllerName = opts.admin ? `modals/${name}` : name;
-    modalController.set("name", controllerName);
+    this.name = controllerName;
 
     let controller = container.lookup("controller:" + controllerName);
     const templateName = opts.templateName || dasherize(name);
 
-    const renderArgs = { into: "modal", outlet: "modalBody" };
+    const renderArgs = { into: "application", outlet: "modalBody" };
```

Comment:
So targeting `application` allows to render into any named outlet? (in this case in a component that's in the application template)

Reply:
Yep, it seems so :+1:

This `.render` call will be removed as part of our Ember 4.x upgrade, so we (hopefully) won't be carrying it for long)